### PR TITLE
Teams: use getAnnotatedTeam, reorganize store

### PR DESCRIPTION
**Pull request by buoyad**
- Remove `getDetails*` actions, replace by `loadTeam` which calls `getAnnotatedTeam`
- Split `TeamMeta` off from `TeamDetails`, with `TeamMeta` driving the team list and `TeamDetails` driving the team page
- Update views

Lots of this is lifted from #21580 

cc @keybase/y2ksquad

### CODE REVIEW

File: shared/actions/teams.tsx [hunk 1]
```diff
@@ -250,8 +250,8 @@ function* inviteByEmail(_: TypedState, action: TeamsGen.InviteToTeamByEmailPaylo
         yield Saga.put(RouteTreeGen.createClearModals())
       }
     }
-    // TODO: refactor invite situation
-    yield Saga.put(TeamsGen.createGetDetails({clearInviteLoadingKey: loadingKey, teamname}))
+    // TODO: make sure this works
+    // yield Saga.put(TeamsGen.createGetDetails({clearInviteLoadingKey: loadingKey, teamname}))
```
Comment (jzila) [inline shared/actions/teams.tsx]: ??

File: shared/actions/teams.tsx [hunk 2]
```diff
@@ -481,13 +484,10 @@ const ignoreRequest = async (action: TeamsGen.IgnoreRequestPayload) => {
       {name: teamname, username},
       Constants.teamWaitingKey(teamname)
     )
-    return false
   } catch (_) {
-    // TODO handle error, but for now make sure loading is unset
-    // TODO get rid of this once core sends us a notification for this (CORE-7125)
-    // TODO just unset loading without calling this
-    return TeamsGen.createGetDetails({teamname}) // getDetails will unset loading
+    // TODO handle error
```
Comment (jzila) [inline shared/actions/teams.tsx]: ??

File: shared/reducers/teams.tsx [hunk 3]
```diff
@@ -79,26 +79,20 @@ export default Container.makeReducer<
...
+  [TeamsGen.teamLoaded]: (draftState, action) => {
+    const {teamID, details} = action.payload
+    draftState.teamDetails.set(teamID, details)
+  },
+  [TeamsGen.setTeamVersion]: (draftState, action) => {
+    const {teamID, version} = action.payload
+    draftState.teamVersion.set(
       teamID,
-      Constants.makeTeamDetails({id: teamID, teamname})
+      Constants.ratchetTeamVersion(version, draftState.teamVersion.get(teamID))
```
Comment (jzila) [inline shared/reducers/teams.tsx]: Why is this happening in the UI instead of coming directly from the service?

File: shared/teams/team/menu-container.tsx [hunk 4]
```diff
@@ -16,15 +16,16 @@ type OwnProps = {
...
-  const isBigTeam = Constants.isBigTeam(state, teamDetails.teamname)
+  const isBigTeam = Constants.isBigTeam(state, teamname)
   return {
     canCreateSubteam: yourOperations.manageSubteams,
     canDeleteTeam: yourOperations.deleteTeam && teamDetails.subteams?.size === 0,
     canManageChat: yourOperations.renameChannel,
     canViewFolder: !yourOperations.joinTeam,
     isBigTeam,
-    teamname: teamDetails.teamname,
+    teamname: teamname,
```
Comment (jzila) [inline shared/teams/team/menu-container.tsx]: can just be `teamname,`

**Comment by jzila** [comment]
Also I tested the basic functionality, but no corner cases.

File: shared/actions/teams.tsx (hunk 1 above)
Comment (buoyad) [reply shared/actions/teams.tsx]: It works :+1:, deleted

File: shared/actions/teams.tsx (hunk 2 above)
Comment (buoyad) [reply shared/actions/teams.tsx]: I removed the part of the comment that isn't relevant anymore

File: shared/reducers/teams.tsx (hunk 3 above)
Comment (buoyad) [reply shared/reducers/teams.tsx]: IIRC the service doesn't maintain state about what the team version is, they come in gregor notifications from the server.

File: shared/actions/teams.tsx [hunk 5]
```diff
@@ -512,106 +512,15 @@ async function createNewTeamFromConversation(
...
 }
 
-function* getDetailsByID(state: TypedState, action: TeamsGen.GetDetailsByIDPayload, logger: Saga.SagaLogger) {
-  const {teamID, _subscribe} = action.payload
-  let teamname = ''
+const loadTeam = async (_: TypedState, action: TeamsGen.LoadTeamPayload, logger: Saga.SagaLogger) => {
+  const {teamID} = action.payload
 
-  // If we're already subscribed to team details for this team ID, we're already up to date
-  const subscriptions = state.teams.teamDetailsSubscriptionCount.get(teamID) ?? 0
```
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/actions/teams.tsx]: Ah, thanks. Will fix.

**Comment by buoyad** [comment]
merged over `TestFastLoaderKeyGen` which is also failing in master
